pkg/security/admission/admission.go

```diff
@@ -177,11 +196,13 @@ func assignSecurityContext(provider scc.SecurityContextConstraintsProvider, pod
 	}
 
 	if len(errs) > 0 {
+		// ensure psc is not mutated if there are errors
+		pod.Spec.SecurityContext = originalPSC
 		return errs
 	}
 
 	// if we've reached this code then we've generated and validated an SC for every container in the
-	// pod so let's apply what we generated
+	// pod so let's apply what we generated.  Note: the psc is already applied.
 	for i, sc := range generatedSCs {
```

Comment:
Why don't I just iterate over the `pod.Spec.Containers`?  That would be more obvious to me as a reader.